prow/github/client.go

```diff
@@ -991,18 +991,30 @@ type githubError struct {
 
 func (c *Client) RemoveLabel(org, repo string, number int, label string) error {
 	c.log("RemoveLabel", org, repo, number, label)
-	ge := githubError{}
-	code, err := c.request(&request{
+	code, body, err := c.requestRaw(&request{
 		method: http.MethodDelete,
 		path:   fmt.Sprintf("%s/repos/%s/%s/issues/%d/labels/%s", c.base, org, repo, number, label),
 		// GitHub sometimes returns 200 for this call, which is a bug on their end.
-		// On 404, do not retry. We will inspect and handle this case appropriately.
-		exitCodes: []int{200, 204, 404},
-	}, &ge)
+		// Do not expect a 404 exit code and handle it separately because we need
+		// to introspect the request's response body.
+		exitCodes: []int{200, 204},
```

Comment:
I think we still want to avoid retries on 404 which is the main purpose this serves. We can selectively unmarshal if we pass `c.request` a [`json.RawMessage`](https://golang.org/pkg/encoding/json/#RawMessage) value and unmarshal further if needed based on the returned code.

Reply:
I think the deleted comment is misleading. 404s are still retried on master in `c.requestRetry`, the only use of exit codes is to return ClientErrors when the return status code is unexpected. Am I missing something?

Reply:
Yeah the comment was wrong. We should probably fix this behavior to avoid wasting tokens retrying anticipated 404s. It looks like we will only retry 404s twice more instead of 7 times more at least.

Reply:
There is a comment related to retrying 404s (Github is not consistent) so I am not sure whether it should be fixed but in a different PR.